src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,46 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<T1, T2> {
+    private T1 first;
+    private T2 second;
 
+    public Pair(T1 first, T2 second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T1 getFirst() {
+        return first;
+    }
+
+    public T2 getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
```

Comment:
what's the purpose of using wildcard here?

Reply:
done